packages/truffle-core/lib/commands/develop.js

```diff
@@ -14,104 +15,79 @@ var command = {
     usage: "truffle develop",
     options: []
   },
-  runConsole: function(config, ganacheOptions, done) {
-    var Console = require("../console");
-    var Environment = require("../environment");
+  runConsole: async (config, ganacheOptions, done) => {
```

Comment:
To avoid dual-wielding concurrency mechanisms, you can get around this pretty easily:

Firstly, remove the `done` argument and instead wrap the whole function with a `util.callbackify()`
```suggestion
  runConsole: util.callbackify(async (config, ganacheOptions) => {
```

Then, anywhere `done` is used in the function, either `return` directly (for success) or `throw` directly (for error).

This preserves the external callback interface while giving us all the benefits of `await`

Reply:
Also wait a second, `await` isn't even used anywhere in this function?

Reply:
⚔️

Reply:
Get 'im outta here!

Reply:
The same with `run`.

Reply:
@gnidan thoughts?

Kinda goes against the `async` everything policy.

Reply:
Yeah but if it is not an async function then there really is no reason to make it a function that returns a Promise.

Reply:
Like, we don't really want everything async. He's referring to methods that need to return Promises. async fundamentally changes the method in that it causes it to then return a Promise.

Reply:
Ok, so two votes to revert `run` & `runConsole` back to sync funcs.

Unless @gnidan stops me, will be pushing the changes shortly.

Reply:
Reverted 😎